subsys/net/l2/openthread/Kconfig

```diff
@@ -130,7 +130,7 @@ config OPENTHREAD_THREAD_PRIORITY
 
 config OPENTHREAD_THREAD_STACK_SIZE
 	int "OpenThread thread stack size"
-	default 6144 if OPENTHREAD_COMMISSIONER
+	default 6144 if OPENTHREAD_COMMISSIONER || OPENTHREAD_JOINER
```

Comment:
@MarekPorwisz is the same amount of memory required for Joiner and Commissioner or can we lower the Joiner requirements?

Reply:
It still failed for me on 4KB and I manged to stop on breakpoints when it was @ ~5KB.

Reply:
What do you think about giving back that ~1 kB of RAM to non-Commissioner devices?

Reply:
If I remember correctly the value was a little bit higher than 5KB and I can not guarantee I got the highest value (I don't even know how did I end up on that breakpoint). I don't think it would be safe to cut more than 512B out of it. We are still working on this software so the values may vary from commit to commit.

Reply:
> What do you think about giving back that ~1 kB of RAM to non-Commissioner devices?

As this value is just a default, you can always adjust it in your application prj.conf file if it feels too high.